kratos/solving_strategies/strategies/explicit_solving_strategy_bfecc.h

```diff
@@ -0,0 +1,546 @@
+
+        LocalSystemVectorType u_copy(dof_size);
+        IndexPartition<SizeType>(r_dof_set.size()).for_each(
+            [&](const SizeType i_dof){
+                const auto it_dof = r_dof_set.cbegin() + i_dof;
+                u_copy[i_dof] = it_dof->GetSolutionStepValue(BufferPosition);
+            }
+        );
+
+        return u_copy;
+    }
+
+    /**
+     * @brief Initialize the BFECC initial forward substep
+     * This method is intended to implement all the operations required before each BFECC initial forward substep
+     */
+    virtual void InitializeBFECCForwardSubstep() {};
+
+    /**
+     * @brief Finalize the BFECC initial forward substep
+     * This method is intended to implement all the operations required after each BFECC initial forward substep
+     */
+    virtual void FinalizeBFECCForwardSubstep() {};
+
+    /**
+     * @brief Initialize the BFECC backward substep
+     * This method is intended to implement all the operations required before each BFECC backward substep
+     */
+    virtual void InitializeBFECCBackwardSubstep() {};
+
+    /**
+     * @brief Finalize the BFECC backward substep
+     * This method is intended to implement all the operations required after each BFECC backward substep
+     */
+    virtual void FinalizeBFECCBackwardSubstep() {};
+
+    /**
+     * @brief Initialize the BFECC final substep
+     * This method is intended to implement all the operations required before each BFECC final substep
+     */
+    virtual void InitializeBFECCFinalSubstep() {};
+
+    /**
+     * @brief Finalize the BFECC final substep
+     * This method is intended to implement all the operations required after each BFECC final substep
+     */
+    virtual void FinalizeBFECCFinalSubstep() {};
+
+    /**
+     * @brief Performs an intermediate RK4 step
+     * This functions performs all the operations required in an intermediate RK4 sub step
+     * @param rFixedDofsValues The vector containing the step n+1 values of the fixed DOFs
+     * @param Substep The type of substep it is
+     */
+    virtual void PerformSubstep(const LocalSystemVectorType& rFixedDofsValues, const Substep SubstepType)
+    {
+        KRATOS_TRY
+
+        // Get the required data from the explicit builder and solver
+        auto& explicit_bs = BaseType::GetExplicitBuilder();
+        auto& r_dof_set = explicit_bs.GetDofSet();
+        const auto& r_lumped_mass_vector = explicit_bs.GetLumpedMassMatrixVector();
+
+        // Get model part and information
+        const double dt = BaseType::GetDeltaTime();
+        KRATOS_ERROR_IF(dt < 1.0e-12) << "ProcessInfo DELTA_TIME is close to zero." << std::endl;
+        auto& r_model_part = BaseType::GetModelPart();
+        auto& r_process_info = r_model_part.GetProcessInfo();
+
+        // Perform the intermidate sub step update
+
+        double time_integration_theta = -1.0;
+        double time_direction = 0.0;
```

Comment:
```suggestion
        double time_integration_theta;
        double time_direction;
```
To make clear that these are initialized below.

Reply:
This may lead to undefined behaviour if we're not careful in the future. I'd personally rather risk having a wrong value than an undeterministic one.

Reply:
In the end I decided to put it all in a struct and have a function initialize the values.